types/react-devtools-inline/commons.d.ts

```diff
@@ -0,0 +1,1303 @@
+        id: number;
+        renderer: ReactRenderer;
+        rendererInterface: RendererInterface;
+    };
+    shutdown: undefined;
+    fastRefreshScheduled: undefined;
+    /** Value: id */
+    'unsupported-renderer-version': number;
+}
+
+export type HookEventListener<EV extends HookEvents> = (
+    params: EV extends keyof HookEventPayload ? HookEventPayload[EV] : unknown,
+) => any;
+
+export interface FiberRootNode {
+    current: FiberNode;
+}
+
+export interface FiberNode {
+    /** First child */
+    child: FiberNode | null;
+    /** Next sibling */
+    sibling: FiberNode | null;
+    /** Parent */
+    return: FiberNode | null;
+    /**
+     * For custom components: the component class or function.
+     * For built-in DOM components: tag name in lower case.
+     * For text: null.
+     */
+    type: React.ComponentType | string | null;
+    /**
+     * For class components: an instance of the component.
+     * For built-in DOM components and text: DOM node.
+     * For FC components: null.
+     */
+    stateNode: React.Component | Element | Text | null;
+    pendingProps: Record<string, unknown> | null;
+    _debugSource: Source | null;
+    key: string | null;
+}
+export interface DevToolsHook {
+    listeners: Record<string, ListenerHandler[]>;
+    rendererInterfaces: Map<RendererID, RendererInterface>;
+    renderers: Map<RendererID, ReactRenderer>;
+    emit(event: HookEvents, data: unknown): void;
+    getFiberRoots: (rendererID: RendererID) => Set<FiberRootNode>;
+    inject: (renderer: ReactRenderer) => number | null;
+    on<EV extends HookEvents>(event: EV, handler: HookEventListener<EV>): void;
+    off<EV extends HookEvents>(event: EV, handler: HookEventListener<EV>): void;
+    sub<EV extends HookEvents>(event: EV, handler: HookEventListener<EV>): () => void;
+    reactDevtoolsAgent?: Record<string, unknown> | null | undefined;
+    /**
+     * Used by react-native-web and Flipper/Inspector
+     */
+    resolveRNStyle?: ResolveNativeStyle | undefined;
+    nativeStyleEditorValidAttributes?: ReadonlyArray<string> | undefined;
+    /** React uses these methods. */
+    checkDCE: (fn: AnyFn) => void;
+    onCommitFiberUnmount: (rendererID: RendererID, fiber: Record<string, unknown>) => void;
+    onCommitFiberRoot: (
+        rendererID: RendererID,
+        /** Added in v16.9 to support Profiler priority labels */
+        fiber: Record<string, unknown>,
+        /** Added in v16.9 to support Fast Refresh */
+        commitPriority?: number,
+        didError?: boolean,
+    ) => void;
+    /** Timeline internal module filtering */
+    getInternalModuleRanges: () => Array<[string, string]>;
+    registerInternalModuleStart: (moduleStartError: Error) => void;
+    registerInternalModuleStop: (moduleStopError: Error) => void;
+    dangerous_setTargetConsoleForTesting?: (fakeConsole: Record<string, unknown>) => void | undefined;
+}
+
+declare var __REACT_DEVTOOLS_GLOBAL_HOOK__: DevToolsHook;
```

Comment:
```suggestion
```
do you need this one? It's not intended to be used via import, just as global via window object (per documentation). That would clean things out, including exclusion for `tslint`?